hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionServerMetrics.java

```diff
@@ -382,6 +385,63 @@ public void testAppend() throws Exception {
     assertCounter("appendNumOps", 73);
   }
 
+  @Test
+  public void testRegionStoreFilesAccessedDaysAndSize() throws Exception {
+    doNPuts(4000, true);
+    admin.flush(tableName);
+    try (RegionLocator locator = connection.getRegionLocator(tableName)) {
+      for (HRegionLocation location : locator.getAllRegionLocations()) {
+        RegionInfo hri = location.getRegion();
+        MetricsRegionAggregateSource agg =
+          rs.getRegion(hri.getRegionName()).getMetrics().getSource().getAggregateSource();
+        String prefix = "namespace_" + NamespaceDescriptor.DEFAULT_NAMESPACE_NAME_STR + "_table_"
+          + tableName.getNameAsString() + "_region_" + hri.getEncodedName() + "_metric_";
+
+        TEST_UTIL.waitFor(60000, 1000, new Waiter.ExplainingPredicate<Exception>() {
+          @Override
+          public boolean evaluate() {
+            return metricsHelper.checkGaugeExists(prefix + "storeFilesAccessed7DaysSize", agg)
+              && metricsHelper.getGaugeLong(prefix + "storeFilesAccessed7DaysSize", agg) > 0;
+          }
+
+          @Override
+          public String explainFailure() {
+            return prefix + "storeFilesAccessed7DaysSize does not exists in region metrics";
+          }
+        });
+
+        metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed7DaysSize", 0, agg);
+        metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed30DaysSize", 0, agg);
+        metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed90DaysSize", 0, agg);
+      }
+    }
+  }
+
+  @Test
+  public void testTableStoreFilesAccessedDaysAndSize() throws Exception {
+    doNPuts(4000, true);
+    admin.flush(tableName);
+    MetricsTableAggregateSource agg = metricsRegionServer.getTableSourceAgg();
+    String prefix = "namespace_" + NamespaceDescriptor.DEFAULT_NAMESPACE_NAME_STR + "_table_"
+      + tableName.getNameAsString() + "_metric_";
+    TEST_UTIL.waitFor(60000, 1000, new Waiter.ExplainingPredicate<Exception>() {
+      @Override
+      public boolean evaluate() {
+        return metricsHelper.checkGaugeExists(prefix + "storeFilesAccessed7DaysSize", agg)
+          && metricsHelper.getGaugeLong(prefix + "storeFilesAccessed7DaysSize", agg) > 0;
+      }
+
+      @Override
+      public String explainFailure() {
+        return prefix + "storeFilesAccessed7DaysSize does not exists in table metrics";
+      }
+    });
+
+    metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed7DaysSize", 0, agg);
+    metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed30DaysSize", 0, agg);
+    metricsHelper.assertGaugeGt(prefix + "storeFilesAccessed90DaysSize", 0, agg);
```

Comment:
[nit] is there any way to verify non-zero metric for `storeFilesAccessed%sDaysSize` in test, e.g. if we can mock the access time ?